packages/salesforce-adapter/test/adapter.discover.test.ts

```diff
@@ -251,6 +252,17 @@ describe('SalesforceAdapter fetch', () => {
       expect(nestedType.fields.doubleNested.type.elemID).toEqual(singleField.elemID)
       expect(singleField).toBeDefined()
       expect(singleField.fields.str.type.elemID).toEqual(BuiltinTypes.STRING.elemID)
+      expect(elementsMap[`salesforce.${INTERNAL_SETTINGS}`]).toBeDefined()
+      expect(elementsMap[`salesforce.${INTERNAL_SETTINGS}.instance`]?.value?.instanceUrl).toBe(MOCK_INSTANCE_URL)
+    })
+
+    it('should not fetch internal settings when url is invalid', async () => {
+      connection.instanceUrl = 'invalidurl'
+      const { elements: result } = await adapter.fetch()
+
+      const elementsMap = _.assign({}, ...result.map(t => ({ [id(t)]: t })))
```

Comment:
nit (I know you copied it from another test, but still)
```suggestion
      const elementsMap = _.keyBy(result, elem => elem.elemID.getFullName())
```
or
```suggestion
      const elementsMap = _.keyBy(result, id)
```